sdk/core/Azure.Core/src/Pipeline/Internal/RequestActivityPolicy.cs

```diff
@@ -123,10 +126,17 @@ private async ValueTask ProcessAsync(HttpMessage message, ReadOnlyMemory<HttpPip
 
             if (message.Response.IsError)
             {
-                scope.AddAttribute("otel.status_code", "ERROR");
-                scope.Failed();
+                if (isActivitySourceEnabled)
+                {
+                    scope.AddAttribute("error.type", statusCodeStr);
+                }
+                else
+                {
+                    scope.AddAttribute("otel.status_code", "ERROR");
```

Comment:
A bit confused why this only applies to diagnostic source

Reply:
`otel.status_code` is a back-compat thing